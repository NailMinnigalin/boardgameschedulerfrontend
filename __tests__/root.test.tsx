import { expect, test } from "vitest";
import { cleanup, render, screen } from "@testing-library/react";

import Page from "../src/app/page";
import { afterEach } from "vitest";

afterEach(() => cleanup());

test("Root page has signIn button", () => {
  render(<Page />);
  expect(screen.getByTestId("signIn_button")).toBeDefined();
});

test("SignIn button has text 'Sign In'", () => {
  render(<Page />);

  let signInButton = screen.getByTestId("signIn_button");

  expect(signInButton.textContent).toEqual("Sign In");
});
