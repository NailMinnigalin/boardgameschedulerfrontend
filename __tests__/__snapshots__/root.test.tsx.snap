// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Snapshot test 1`] = `
<DocumentFragment>
  <div
    class="flex min-h-screen flex-col items-center justify-center p-4"
  >
    <h1
      class="mb-6 text-3xl font-bold"
    >
      Board Game Scheduler
    </h1>
    <div
      class="space-y-4"
    >
      <a
        class="inline-flex items-center justify-center gap-2 whitespace-nowrap text-sm font-medium transition-[color,box-shadow] disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 [&_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive text-primary-foreground shadow-xs h-9 has-[>svg]:px-3 rounded border border-gray-500 bg-gray-800 px-4 py-2 hover:bg-gray-700"
        data-slot="button"
        data-testid="signIn_button"
        href="/signin"
      >
        Sign In
      </a>
    </div>
  </div>
</DocumentFragment>
`;
